lib/modules/pipeline/webpack.config.js

```diff
@@ -1,27 +1,36 @@
-// some packages, plugins, and presets referenced/required in this webpack
-// config are deps of embark and will be transitive dapp deps unless specified
-// in the dapp's own package.json
-
-// embark modifies process.env.NODE_PATH so that when running dapp scripts in
-// embark's child processes, embark's own node_modules directory will be
-// searched by node's require(); however, webpack and babel do not directly
-// support NODE_PATH, so modules such as babel plugins and presets must be
-// resolved with require.resolve(); that is only necessary if a plugin/preset
-// is in embark's node_modules vs. the dapp's node_modules
+/* global __dirname module process require */
 
-const cloneDeep = require('lodash.clonedeep');
-// const CompressionPlugin = require('compression-webpack-plugin');
-const glob = require('glob');
-const HardSourceWebpackPlugin = require('hard-source-webpack-plugin');
 const path = require('path');
 
 const dappPath = process.env.DAPP_PATH;
 const embarkPath = process.env.EMBARK_PATH;
 
+const dappNodeModules = path.join(dappPath, 'node_modules');
+const embarkNodeModules = path.join(embarkPath, 'node_modules');
+
+function customRequire(mod) {
+  return require(customRequire.resolve(mod));
+}
+
+customRequire.resolve = function (mod) {
+  return require.resolve(
+    mod,
+    {paths: [dappNodeModules, embarkNodeModules]}
```

Comment:
Just for my own understanding, isn't it a dangerous pattern.
e.g the dapp depends of embarkNodeModules.
We update the version and it breaks the dapps

What do you think?

Reply:
It is a risk, one that we've had all along — it is a consequence of depending upon a global installation of embark to, effectively, provide transitive dependencies to a DApp.

The behavior of `customRequire() / customRequire.resolve()` in this PR is identical with the behavior `NODE_PATH` has provided since it was introduced in #720. It wasn't necessary to do that kind of thing prior to #720 since, up to that point, embark's child processes running in cwd (top-level DApp directory) didn't attempt to load any modules in embark's `node_modules`. An *ejected* webpack config does attempt to do that. By appending embark's node_modules to `NODE_PATH`, it can work. If I had the idea of `customRequire()` at the time, I would have implemented that instead.

The reason we need a different mechanism for this PR is that `NODE_PATH` is set on `process.env` in the main embark process (in `cmd.js`), but that only affects child processes (e.g. the one running webpack). @emizzle's change detection code loads the config in the main process and therefore it was breaking.

Another option would be to require DApp's to provide all the babel and webpack related dependencies. A common mechanism to introduce that kind of requirement is peer dependencies (in embark's `package.json`). However, at a tooling level (command-line `npm`) that doesn't work for globally installed packages; it would only be practical if embark was intended (and documented) to be installed as a local dependency of a DApp.

I'm definitely open to ideas about how we can handle this better.

Reply:
Thx for clarification. I guess the best for us is as you said to move forward on the local embark